pkg/localcommit/localcommit.go

```diff
@@ -0,0 +1,91 @@
+package localcommit
+
+import (
+	"bytes"
+	"path/filepath"
+
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/go-openapi/strfmt"
+)
+
+type FileDoesNotExistError struct{ *locale.LocalizedError }
+
+func IsFileDoesNotExistError(err error) bool {
+	return errs.Matches(err, &FileDoesNotExistError{})
+}
+
+func Get(projectDir string) (string, error) {
+	configDir := filepath.Join(projectDir, constants.ProjectConfigDirName)
+	commitFile := filepath.Join(configDir, constants.CommitIdFileName)
+	if !fileutils.DirExists(configDir) || !fileutils.TargetExists(commitFile) {
+		return "", &FileDoesNotExistError{locale.NewError("err_commit_file_does_not_exist",
+			"Your project runtime's commit ID file '{{.V0}}' does not exist", commitFile)}
+	}
+
+	b, err := fileutils.ReadFile(commitFile)
+	if err != nil {
+		return "", locale.WrapError(err, "err_get_commit_file", "Could not read your project runtime's commit ID file")
+	}
+
+	commitID := string(b)
+	if !strfmt.IsUUID(commitID) {
+		return "", locale.NewError("err_commit_id_invalid", commitID)
+	}
+
+	return commitID, nil
+}
+
+func GetUUID(projectDir string) (strfmt.UUID, error) {
+	commitID, err := Get(projectDir)
+	if err != nil {
+		return "", errs.Wrap(err, "Unable to get local commit")
+	}
+	return strfmt.UUID(commitID), nil
+}
+
+func Set(projectDir, commitID string) error {
+	if !strfmt.IsUUID(commitID) {
+		return locale.NewError("err_commit_id_invalid", commitID)
+	}
+
+	commitFile := filepath.Join(projectDir, constants.ProjectConfigDirName, constants.CommitIdFileName)
```

Comment:
nit: You're doing this in `Get()` as well. Suggest making it a private function.